### PR TITLE
Update Loyalty API to deploy the Account CloudWatchRole that is required

### PR DESCRIPTION
**Pull request by deldrid1**
*Issue #, if available:*
#128

*Description of changes:*
I ran into the same issue described in #128.  

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-account.html holds the answer to fix this programatically and in a region agnostic way.

FYI - There seems to be some fragility with the order of operations here:
image

I haven't tested on a clean AWS account, but given that Payments API deploys clean, I think this fix will work (I'm not a SAM expert and am coming from another cloud provider so troubleshooting this has been quite the learning experience!)

Note: if you're in the state described in #128, you will need to manually go into the CloudFormation Console and delete the stack stuck in `ROLLBACK_COMPLETE` or you will get an error like this one: 
  - image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by heitorlessa** [comment]
Hey Austin - Thanks for taking the time to do this.

I'll go through this during the week, and circle back here :)

I was nearly sure this was in place already. I'll do some digging in the history to double check before I make some minor suggestions here

**Comment by heitorlessa** [comment]
Just remember why this wasn't added now - API Gateway only allows a single IAM Role, so if customers were to deploy this in their accounts it can override a previously setup IAM Role in their accounts (need to double check the behaviour).

Regardless, we should provide these resources commented for educational purposes, and remove the attempt to have Logs enabled as it currently stands to prevent build failures.

Could you allow me to make changes in your fork so I can change the naming convention, and add some info there?

https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

**Comment by deldrid1** [comment]
Looks checked to me? I didn’t make any changes

**Comment by heitorlessa** [comment]
Great, I’ll push changes this week and ask for your review

**Comment by heitorlessa** [comment]
I'm getting access denied when trying to push to your fork/branch w.r.t this PR - I've created a patch file that you can apply, so I can merge it. 

https://gist.github.com/heitorlessa/515584f84d40208d6e4c329d78fbb691

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Apply patch (`git apply <patch>`): https://gist.github.com/heitorlessa/515584f84d40208d6e4c329d78fbb691

**Comment by deldrid1** [comment]
Not sure what is going on with the permissions...
image


I've applied the patch.  From my perspective this is good to go, although it will be a breaking change for folks that blindly pull with a working repo - logs that were there will no longer be configured after a fresh deply so you may want to think about that in terms of versioning.  

I would also recommend some additional README content somewhere about the ability to enable this logging and why its useful, but its your repo so I'll leave it to you to decide if that is useful or not!

**Comment by heitorlessa** [comment]
Appreciate the changes and comments @deldrid1, truly. As for the docs, I'll do this separately, specifically the [decision log](https://github.com/aws-samples/aws-serverless-airline-booking/blob/develop/src/backend/loyalty/README.md#decisions-log) section in Loyalty to describe why that was necessary. 

Significant changes to the project per se are also documented here: https://github.com/aws-samples/aws-serverless-airline-booking/blob/develop/docs/decision_log.md

After re:Invent, I'll get back to this to tackle some major operations, testing, etc. documented in the Boards/Issues.

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**
